eZ/Publish/Core/Pagination/Pagerfanta/AdapterFactory/SearchHitAdapterFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Pagination\Pagerfanta\AdapterFactory;
+
+use eZ\Publish\API\Repository\SearchService;
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\Core\Pagination\Pagerfanta\ContentSearchHitAdapter;
+use eZ\Publish\Core\Pagination\Pagerfanta\LocationSearchHitAdapter;
+use Pagerfanta\Adapter\AdapterInterface;
+use Pagerfanta\Adapter\FixedAdapter;
+
+/**
+ * @internal
+ */
+final class SearchHitAdapterFactory implements SearchHitAdapterFactoryInterface
+{
+    /** @var \eZ\Publish\API\Repository\SearchService */
+    private $searchService;
+
+    public function __construct(SearchService $searchService)
+    {
+        $this->searchService = $searchService;
+    }
+
+    public function createAdapter(Query $query, array $languageFilter = []): AdapterInterface
+    {
+        if ($query instanceof LocationQuery) {
+            return new LocationSearchHitAdapter($query, $this->searchService, $languageFilter);
+        } else {
```

Comment:
not rly needed

Reply:
Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/2970/commits/c0d91a5c05a1e8c6c92097b16e9555ee55d76189